lib/io/hardware_spi.cpp

```diff
@@ -118,19 +118,19 @@ const char *HardwareSpi::getSpiBusAddress(const SpiBus bus)
   }
 }
 
-std::uint32_t HardwareSpi::getClockValue(Clock clock)
+std::uint32_t HardwareSpi::getClockValue(SpiClock clock)
 {
   switch (clock) {
-    case Clock::k500KHz:
-      return 500000;
-    case Clock::k1MHz:
-      return 1000000;
-    case Clock::k4MHz:
-      return 4000000;
-    case Clock::k16MHz:
-      return 16000000;
-    case Clock::k20MHz:
-      return 20000000;
+    case SpiClock::k500KHz:
```

Comment:
I know this is one of my more pedantic comments (and that's really saying something) bit will there ever be a case where the number of `SpiClock` members changes? I can currently deal with the number of cases in this switch statement but if this has score to expand then maybe a different method of handling this is more appropriate?

Reply:
no I don't think so? though I still haven't tested spi this year, last year these values worked just fine